### ISSUE TITLE
Host.json configuration should be grouped

### ISSUE THREAD

**Issue by cgillum** [Azure/azure-functions-durable-extension#641]
The number of durable-specific settings in host.json are fairly large. We should group them to make it easier to continue adding more.

There's also a desire to be able to choose different storage providers, in which case provider-specific settings will need to be distinguished from each other.  A (brainstorm) proposal:

```json
{
  "durableTask": {
    "hubName": "MyTaskHub",
    "provider": {
        "azureStorage": {
            "controlQueueBatchSize": 32,
            "partitionCount": 4,
            "controlQueueVisibilityTimeout": "00:05:00",
            "workItemQueueVisibilityTimeout": "00:05:00",
            "maxConcurrentActivityFunctions": 10,
            "maxConcurrentOrchestratorFunctions": 10,
            "connectionStringName": "AzureWebJobsStorage",
        }
    },
    "tracing": {
        "traceInputsAndOutputs": false,
        "traceReplayEvents": false,
    },
    "notifications": {
        "eventGrid": {
            "topicEndpoint": "https://topic_name.westus2-1.eventgrid.azure.net/api/events",
            "keySettingName":  "EventGridKey",
            "publishRetryCount": 3,
            "publishRetryInterval": "00:00:30",
            "publishEventTypes": ["Started", "Pending", "Failed", "Terminated"]
        }
    }
  }
}
```

**Comment by CarlosSardo** [comment]
Hi @cgillum 

I would like to help out by picking up this issue, if possible.

Thanks!

**Comment by cgillum** [comment]
@CarlosSardo that would be super helpful! Connor has already done the storage provider part but I would love help with the others. If you'd like to submit a PR, this work should currently target the v2 branch.

**Comment by CarlosSardo** [comment]
@cgillum @ConnorMcMahon 

Here's a [draft PR](https://github.com/CarlosSardo/azure-docs/pull/1) to the [preview docs](https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-preview#hostjson-schema), regarding this breaking change. It can be submitted, once you decide to ship it.

**Comment by cgillum** [comment]
This change has been merged and will go out with our `2.0.0-beta1` release.

**Comment by CarlosSardo** [comment]
The related PR with updates to the **official docs** can be found [here](https://github.com/MicrosoftDocs/azure-docs/pull/34753).
